src/stream/src/executor/actor.rs

```diff
@@ -87,10 +88,13 @@ impl OperatorInfoStatus {
     }
 }
 
-/// Shared by all operators in the stream.
+/// Shared by all operators of an actor.
 #[derive(Default)]
 pub struct ActorContext {
     pub info: Vec<OperatorInfo>,
+
+    /// TODO: report errors and prompt the user.
+    pub errors: HashMap<String, Vec<ExprError>>,
```

Comment:
How about locking the inner field instead of the whole context? So that we may put more fields inside and acquire them without locking. (only after we make the context truly actor-local)

```rust
struct Context {
  id: ActorId,
  errors: Mutex<HashMap<_, _>>,
}

type ContextRef = Arc<Context>;
```

Reply:
BTW, what do you mean by "truly actor-local"? 🤔

Reply:
The current implementation also looks good to me.

Reply:
This struct is shared by all actors in the compute node now. 🥵🥵🥵

Reply:
Ohhhhh. That would be a big problem 😇

Reply:
> This struct is shared by all actors in the compute node now. 🥵🥵🥵

Are you sure? 🤔 I think it's already actor-local

```rs
fn build_actors(&mut self, actors: &[ActorId], env: StreamEnvironment) -> Result<()> {
    for &actor_id in actors {
        let actor = self.actors.remove(&actor_id).unwrap();
        let actor_context = Arc::new(Mutex::new(ActorContext::default()));
        
		...

		let actor = Actor::new(
            dispatcher,
            actor_id,
            self.context.clone(),
            self.streaming_metrics.clone(),
            actor_context,
        );
	}
}
```

Reply:
Do you mean `SharedContext` instead of `ActorContext`? 🥵

Reply:
My bad. Passing a `Mutex` wrapped context to the construction function really brings a intuition of that to me. 😢🥵

Reply:
I will try refactoring in a separate PR for ease of review